### ISSUE TITLE
bilibili无法下载

### ISSUE THREAD

**Issue by nqzhang** [soimort/you-get#1952]
Please make sure these boxes are checked before submitting your issue – thank you!

- [ ] You can actually watch the video in your browser or mobile application, but not download them with `you-get`.
- [ ] Your `you-get` is up-to-date.
- [ ] I have read <https://github.com/soimort/you-get/wiki/FAQ> and tried to do so.
- [ ] The issue is not yet reported on <https://github.com/soimort/you-get/issues> or  <https://github.com/soimort/you-get/wiki/Known-Bugs>. If so, please add your comments under the existing issue.
- [ ] The issue (or question) is really about `you-get`, not about some other code or project.

Run the command with the `--debug` option, and paste the full output inside the fences:

```
[PASTE IN ME]
```

If there's anything else you would like to say (e.g. in case your issue is not about downloading a specific video; it might as well be a general discussion or proposal for a new feature), fill in the box below; otherwise, you may want to post an emoji or meme instead:

> [WRITE SOMETHING]
> [OR HAVE SOME :icecream:!]

汉语翻译最终日期：2016年02月26日

在提交前，请确保您已经检查了以下内容!

- [ ] 你可以在浏览器或移动端中观看视频，但不能使用`you-get`下载.
- [ ] 您的`you-get`为最新版.
- [ ] 我已经阅读并按 <https://github.com/soimort/you-get/wiki/FAQ> 中的指引进行了操作.
- [ ] 您的问题没有在<https://github.com/soimort/you-get/issues> , <https://github.com/soimort/you-get/wiki/FAQ> 或 <https://github.com/soimort/you-get/wiki/Known-Bugs> 报告，否则请在原有issue下报告.
- [ ] 本问题确实关于`you-get`, 而不是其他项目.

请使用`--debug`运行，并将输出粘贴在下面:

```
you-get --debug https://www.bilibili.com/video/av10390679/
[DEBUG] get_content: http://www.bilibili.com/video/av10390679/
[DEBUG] get_content: http://interface.bilibili.com/playurl?&cid=17161200&from=miniplay&player=1&sign=35a69bfac3fd72799086844962835d3d
[DEBUG] url_info: http://ws.acgvideo.com/9/9e/17161200-1.flv?wsTime=1494221135&platform=pc&wsSecret2=d883850708bcf63bc497a319d2d296c5&oi=1158437299&rate=6
you-get: version 0.4.715, a tiny downloader that scrapes the web.
you-get: ['https://www.bilibili.com/video/av10390679/']
Traceback (most recent call last):
  File "/usr/bin/you-get", line 11, in <module>
    sys.exit(main())
  File "/usr/lib/python3.4/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1429, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1342, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1158, in download_main
    download(url, **kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1422, in any_download
    m.download(url, **kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/extractors/bilibili.py", line 187, in bilibili_download
    info_only=info_only)
  File "/usr/lib/python3.4/site-packages/you_get/extractors/bilibili.py", line 101, in bilibili_download_by_cid
    _, type_, temp = url_info(url)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 425, in url_info
    response = urlopen_with_retry(request.Request(url))
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 310, in urlopen_with_retry
    return request.urlopen(*args, **kwargs)
  File "/usr/lib64/python3.4/urllib/request.py", line 161, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.4/urllib/request.py", line 470, in open
    response = meth(req, response)
  File "/usr/lib64/python3.4/urllib/request.py", line 580, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python3.4/urllib/request.py", line 502, in error
    result = self._call_chain(*args)
  File "/usr/lib64/python3.4/urllib/request.py", line 442, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.4/urllib/request.py", line 685, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/usr/lib64/python3.4/urllib/request.py", line 470, in open
    response = meth(req, response)
  File "/usr/lib64/python3.4/urllib/request.py", line 580, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python3.4/urllib/request.py", line 508, in error
    return self._call_chain(*args)
  File "/usr/lib64/python3.4/urllib/request.py", line 442, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.4/urllib/request.py", line 588, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden

```

如果您有其他附言，例如问题只在某个视频发生，或者是一般性讨论或者提出新功能，请在下面添加;或者您可以卖个萌:

> [您的内容]
> [舔 :icecream:!]

**Comment by rosynirvana** [comment]
You can try #1698 or the develop branch

**Comment by nqzhang** [comment]
develop分支可以

**Comment by nqzhang** [comment]
发现还是有些视频无法下载
 you-get --debug http://www.bilibili.com/video/av10288909
[DEBUG] get_content: http://www.bilibili.com/video/av10288909
[DEBUG] get_content: http://interface.bilibili.com/playurl?&cid=16998278&from=miniplay&player=1&sign=9014bc146ee110c385bc6a0a4f308057
Site:       bilibili.com
Title:      【拜托了冰箱 第三季】第4期：安吉吐槽沙溢磕碜，李湘认安吉做干儿子
Type:       Unknown type ()
Size:       0.0 MiB (0 Bytes)

you-get: version 0.4.715, a tiny downloader that scrapes the web.
you-get: ['http://www.bilibili.com/video/av10288909']
Traceback (most recent call last):
  File "/usr/bin/you-get", line 11, in <module>
    load_entry_point('you-get==0.4.715', 'console_scripts', 'you-get')()
  File "/usr/lib/python3.4/site-packages/you_get/__main__.py", line 92, in main
    main(**kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1443, in main
    script_main('you-get', any_download, any_download_playlist, **kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1354, in script_main
    download_main(download, download_playlist, args, playlist, output_dir=output_dir, merge=merge, info_only=info_only, json_output=json_output, caption=caption)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1156, in download_main
    download(url, **kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 1436, in any_download
    m.download(url, **kwargs)
  File "/usr/lib/python3.4/site-packages/you_get/extractors/bilibili.py", line 187, in bilibili_download
    info_only=info_only)
  File "/usr/lib/python3.4/site-packages/you_get/extractors/bilibili.py", line 106, in bilibili_download_by_cid
    download_urls(urls, title, type_, total_size=None, output_dir=output_dir, merge=merge, timeout=1, headers={'Referer': 'http://www.bilibili.com/'})
  File "/usr/lib/python3.4/site-packages/you_get/common.py", line 761, in download_urls
    assert urls
AssertionError

**Comment by rosynirvana** [comment]
因为这个用的不是bilibili的接口，也不是bilibili的播放器
```
curl 'http://interface.bilibili.com/player?id=cid:16998278' | grep 'oriurl'
```

**Comment by nqzhang** [comment]
原来这样 谢谢

**Comment by rosynirvana** [comment]
现在#1698 支持bilibili的腾讯播放器视频
```
./you-get -di  http://www.bilibili.com/video/av10288909
[DEBUG] url_locations: http://www.bilibili.com/video/av10288909
[DEBUG] get_content: http://www.bilibili.com/video/av10288909/
[DEBUG] get_content: http://interface.bilibili.com/player?id=cid:16998278
[DEBUG] url_locations: http://v.qq.com/page/p/0/0/p0023iyz6w9.html
[DEBUG] get_response: https://v.qq.com/x/page/p0023iyz6w9.html
[DEBUG] get_response: http://vv.video.qq.com/getinfo?otype=json&appver=3%2E2%2E19%2E333&platform=11&defnpayver=1&vid=p0023iyz6w9
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.1.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.1.mp4?vkey=9A4F8AE3800B7E034C495F4C514668BCD4A633DDD8006B7DBF2793451EBF1019316F942CEB0F96087D6DEC58F8C22371A9C6EF42A546C57B69BACF275ABDC204464A2DB8E2CB9037CC093CC7ED4798359C856059E94D2D4A65366AE88B6E0511
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.2.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.2.mp4?vkey=499696D5B6F51E4221D6E3A577DD797EA9BF86ABFE236109166BEC96DB0243AA41DED06162759527D5CA3F21F88C1EEFE6C6DC8D16EF45EC4A140192EB29E935F6BAAEC96F5C7BB4CD3BCC365860B7411887F6E164955F3F1CB38EA4127FEB5F
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.3.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.3.mp4?vkey=6F29A22F000FEFBBA90CCA2ACEBD941C069CFA9E64E33AF6CB3C294A75A004FA7D39CAD1C30C0455DC861A48B0EAB73D68E3718EAA8DF85FCA859F76CE2BB6B4ABAACFEBB3BEEC7306A92C4980593CB16A3E0BFF8A3D5A336F817190376E6B60
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.4.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.4.mp4?vkey=69E7236CEE16EFDF8B08AB7D35814CBC555BEFBA51FC612736AEA8D33F90DFCC90E5D7957830CF96350F2CC822C0D0DA304A2FEE38AF98D1421759C07F1A18500268CCD10B5276858E67ED3F3E3C7F3D77FFFC4460CCBD61FF97F15F9FB3D1E2
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.5.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.5.mp4?vkey=150CB26965EE16D5205ECB95F3ED6CA8E0843171BA474203B9C374947F69B23AC87611713CD6210EF4ECE86A1BF237F0D3366011D25709D094F11C47292A78C139FD0CE85F960779EAA869FEB4629133E6FC9F99E0D6AB53B956DE79AC910973
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.6.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.6.mp4?vkey=50F086FEF89D0ECE085B13B120BE2C86BC68F1952E0618137AD41CF7DC0452A0C77933BAB96DC3A198F6A13F50A3F53A8D9CAAACA01B7B635D4990D054F71F2276B9DB0CFD5DD19EE925DF5823553F161A0458F5CF7F1E62DE6C9E69A55E770B
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.7.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.7.mp4?vkey=AFBC4DA1C3012BC4C9B8208CA103266BE8A6A23C324959F4057AE388CFD4874A2EEE9228CFFD83C08B8265CDE97B67BB366B0E03C04370BAF4664ACC2EF63EDC7449112D224E2032BF5D12DC2DA7F4197E173AFF719BE8A6260181D835CBAADC
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.8.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.8.mp4?vkey=BCE6AF0D73FA77004BD7F6B2F6CAFD3B4E4B4DDAC086770F8E43C663CF11E74BFB318D76CDBF5ABCBB620DAA45954637B411F5EF83FB0FFC01893CFFA6729719C3EB7A4049B2140555A47D052239632B4A5B596DB6FE818BE12070AACDBD9DB1
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.9.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.9.mp4?vkey=201292D5909E8C0BB5B1C941B657BB9EC7DCB95970C7D81DB8AF0D586560CFCB5C863A58EF4829A3211D8BF558D18AAC65AAD848CAEB4A1A77E3BB701D40522A94BAFF2AF6F94AF251C90AA7CD753D54A4BF2CA062FE77F4BEB3AF87928C3AFF
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.10.mp4
[DEBUG] url_info: http://vhoth.dnion.videocdn.qq.com/flv/123/183/p0023iyz6w9.p209.10.mp4?vkey=4E497193F8C19429011752C4F00B5DD0F6C9645B51ED569B226462C70FEF5D41CA054A19DC741578FA0EE4F6F7E318FFE5FA4382C5D0D604CDE6DCFF3E2B461221C4E75417C56341B1C814FCCAB4F2D0E788256FADEA82B756C78BB94927BA35
[DEBUG] get_response: http://vv.video.qq.com/getkey?otype=json&platform=11&format=10209&vid=p0023iyz6w9&filename=p0023iyz6w9.p209.11.mp4
Site:       QQ.com
Title:      第4期：安吉吐槽沙溢磕碜，李湘认安吉做干儿子
Type:       MPEG-4 video (video/mp4)
Size:       1045.18 MiB (1095954078 Bytes)
```

**Comment by soimort** [comment]
Closed via #1698.
